src/awkward/operations/ak_concatenate.py

```diff
@@ -152,7 +152,7 @@ def _impl(arrays, axis, mergebool, highlevel, behavior, attrs):
         batches = [[content_or_others[0]]]
         for x in content_or_others[1:]:
             batch = batches[-1]
-            if ak._do.mergeable(batch[-1], x, mergebool=mergebool):
+            if ak._do.mergeable(batch[0], x, mergebool=mergebool):
```

Comment:
```suggestion
            if ak._do.mergeable(batch[-1], x, mergebool=mergebool):
```

Although `mergeable` is an equivalence relation and we can compare a new item with any of the already established items, I think it's easier to think about as always checking the last one. (If you were to draw a diagram of it, the lines wouldn't cross.)